File: java/backend/src/main/java/com/example/serviceplatform/domain/LeadRequest.java
package com.example.serviceplatform.domain;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import java.time.Instant;

public record LeadRequest(
    String id,
    @NotBlank String name,
    @Email @NotBlank String email,
    @NotBlank String phone,
    @NotBlank String service,
    @NotBlank String message,
    Instant createdAt) {}

File: rust/backend/src/model/lead_request.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;

#[derive(Clone, Debug, sqlx::FromRow)]
pub struct LeadRequest {
    pub id: Uuid,
    pub name: String,
    pub email: String,
    pub phone: String,
    pub service: String,
    pub message: String,
    pub created_at: DateTime<Utc>,
}
